src/lib/mina_base/transaction_logic.ml

```diff
@@ -1234,7 +1247,7 @@ module Make (L : Ledger_intf) : S with type ledger := L.t = struct
             succeeds, to ensure that the network is compensated for processing this
             command.
           *)
-          set_with_location ledger loc acct' ;
+          ok_or_reject @@ set_with_location ledger loc acct' ;
```

Comment:
should `ok_or_reject` be modified to return an `Or_error.t` instead of raising? Transaction application would fail  otherwise (due to a full ledger or other errors), right?

Reply:
I think this `ok_or_reject` already does that:
```ocaml
        let ok_or_reject = function
          | Ok x ->
              x
          | Error err ->
              return (Error err)
        in
```
where `return` is from the wrapping `with_return (fun {return} -> ... )`.

Reply:
There's another one defined here https://github.com/MinaProtocol/mina/blob/714fcfb48cee638b4538ed3436a8a0ced9ddccbb/src/lib/mina_base/transaction_logic.ml#L668 in `apply_user_command_unchecked` function